internal/services/streamanalytics/stream_analytics_output_powerbi_resource_test.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/streamanalytics/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type StreamAnalyticsOutputPowerBIResource struct{}
+
+func (r StreamAnalyticsOutputPowerBIResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.OutputID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.StreamAnalytics.OutputsClient.Get(ctx, id.ResourceGroup, id.StreamingjobName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func TestAccStreamAnalyticsOutputPowerBI_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_stream_analytics_output_powerbi", "test")
+	r := StreamAnalyticsOutputPowerBIResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStreamAnalyticsOutputPowerBI_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_stream_analytics_output_powerbi", "test")
+	r := StreamAnalyticsOutputPowerBIResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config: r.updated(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r StreamAnalyticsOutputPowerBIResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_stream_analytics_output_powerbi" "test" {
+  name                      = "acctestoutput-%d"
+  stream_analytics_job_name = azurerm_stream_analytics_job.test.name
+  resource_group_name       = azurerm_stream_analytics_job.test.resource_group_name
+  dataset                   = "foo"
+  table                     = "bar"
+  group_id                  = "85b3dbca-5974-4067-9669-67a141095a76"
+  group_name                = "some-test-group-name"
+  authentication_mode       = "Msi"
+}
+`, template, data.RandomInteger)
+}
+
+func (r StreamAnalyticsOutputPowerBIResource) updated(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_stream_analytics_output_powerbi" "test" {
+  name                      = "acctestoutput-%d"
+  stream_analytics_job_name = azurerm_stream_analytics_job.test.name
+  resource_group_name       = azurerm_stream_analytics_job.test.resource_group_name
```

Comment:
could these be combined
```suggestion
  stream_analytics_job_id = azurerm_stream_analytics_job.test.id
```